python/tchannel/scheme.py

```diff
@@ -41,10 +62,18 @@ def __init__(self):
     def type(self):
         return self._type
 
-    def deserialize(self, obj):
+    def deserialize_header(self, obj):
+        assert obj
```

Comment:
I'm not suggesting you treat it as an error. Treat it as valid input. Make `deserialize(None) == deserialize('') == None`